### PR TITLE
Image

### PR DESCRIPTION
**Pull request by LjiljanaN**

### CODE REVIEW

File: HRTool/src/main/java/app/controllers/ImageUploadRestController.java [hunk 1]
```diff
@@ -17,40 +15,17 @@
...
-
-
-	}
 	
 	@RequestMapping(value = "/upload", method = RequestMethod.POST)
-	public @ResponseBody String uploadingPOST(@RequestParam("flowFilename") String fileName,
-			@RequestParam("flowTotalChunks") int totalChunks,
-			@RequestParam("slika") MultipartFile file){
-		System.out.println("----------------POST-------------");
+	public @ResponseBody String uploadingPOST(@RequestParam("picture") MultipartFile file){
```
Comment (cikaRadule) [inline HRTool/src/main/java/app/controllers/ImageUploadRestController.java]: @ResponseBody se podrazumeva, nije potrebno navoditi ga

File: HRTool/src/main/java/app/controllers/ImageUploadRestController.java [hunk 2]
```diff
@@ -17,40 +15,17 @@
...
 		if (!file.isEmpty()) {
-			System.out.println("nije prazan  "+ fileName+ "  totalchunks: "+totalChunks);
             try {
```
Comment (cikaRadule) [inline HRTool/src/main/java/app/controllers/ImageUploadRestController.java]: Formatiran kod nije. Doterati ga moras...

File: HRTool/src/main/java/app/controllers/ImageUploadRestController.java (hunk 1 above)
Comment (LjiljanaN) [inline HRTool/src/main/java/app/controllers/ImageUploadRestController.java]: Ne radi bez te anotacije

File: HRTool/src/main/java/app/controllers/ImageUploadRestController.java [hunk 3]
```diff
@@ -14,55 +12,26 @@
...
 
-
-	}
-	
 	@RequestMapping(value = "/upload", method = RequestMethod.POST)
-	public @ResponseBody String uploadingPOST(@RequestParam("flowFilename") String fileName,
-			@RequestParam("flowTotalChunks") int totalChunks,
-			@RequestParam("slika") MultipartFile file){
-		System.out.println("----------------POST-------------");
+	public @ResponseBody String uploadingPOST(
```
Comment (nantic) [inline HRTool/src/main/java/app/controllers/ImageUploadRestController.java]: [at]ResponseBody is not needed. Any method in the [at]RestController will automatically get it.
Do not return String. Return ResponseEntity.

File: HRTool/src/main/java/app/controllers/ImageUploadRestController.java [hunk 4]
```diff
@@ -14,55 +12,26 @@
 import org.springframework.web.bind.annotation.ResponseBody;
 import org.springframework.web.multipart.MultipartFile;
 
-
-
 @Controller
```
Comment (nantic) [inline HRTool/src/main/java/app/controllers/ImageUploadRestController.java]: This needs to be [at]RestContoller, not [at]Controller

Comment (nantic) [inline HRTool/src/main/java/app/controllers/ImageUploadRestController.java]: This needs to be [at]RestContoller, not [at]Controller

File: HRTool/src/main/java/app/controllers/ImageUploadRestController.java [hunk 5]
```diff
@@ -14,55 +12,26 @@
...
-
 public class ImageUploadRestController {
```
Comment (nantic) [inline HRTool/src/main/java/app/controllers/ImageUploadRestController.java]: This controller shouldn't exist. Methods in this controller should belong to a resource

File: HRTool/src/main/java/app/controllers/ImageUploadRestController.java [hunk 6]
```diff
@@ -14,55 +12,26 @@
...
-		
-		
-		System.out.println("---------------------------------");
-		return "BLA BLA";
+	private ServletContext servletContext;
 
-
-	}
-	
 	@RequestMapping(value = "/upload", method = RequestMethod.POST)
```
Comment (nantic) [inline HRTool/src/main/java/app/controllers/ImageUploadRestController.java]: RequestMapping URL should target resources, never actions/methods.
Ex. Uploading an image for an employee would target /employees/{employeeId}/image.

File: HRTool/src/main/java/app/controllers/ImageUploadRestController.java [hunk 7]
```diff
@@ -14,55 +12,26 @@
...
-                System.out.println("---------------FILE UPLOADED------------" + servletContext.getRealPath("/images"));
-                return "images/"+fileName;
-            } catch (Exception e) {
-                return "You failed to upload " + e.getMessage();
-            }
-        } else {
-            return "You failed to upload  because the file was empty.";
-        }
+			try {
+				file.transferTo(new File(servletContext.getRealPath("/images")
```
Comment (nantic) [inline HRTool/src/main/java/app/controllers/ImageUploadRestController.java]: All constant strings should be extracted.
Ex. private static final String IMAGE_PATH = "/images";

File: HRTool/src/main/java/app/controllers/ImageUploadRestController.java [hunk 8]
```diff
@@ -14,55 +12,26 @@
...
-
-	}
-	
 	@RequestMapping(value = "/upload", method = RequestMethod.POST)
-	public @ResponseBody String uploadingPOST(@RequestParam("flowFilename") String fileName,
-			@RequestParam("flowTotalChunks") int totalChunks,
-			@RequestParam("slika") MultipartFile file){
-		System.out.println("----------------POST-------------");
+	public @ResponseBody String uploadingPOST(
+			@RequestParam("picture") MultipartFile file) {
```
Comment (nantic) [inline HRTool/src/main/java/app/controllers/ImageUploadRestController.java]: If you're uploading a file then the request param should be "file". If you're uploading and image then the object name should be "image".

File: HRTool/src/main/java/app/controllers/ImageUploadRestController.java [hunk 9]
```diff
@@ -14,55 +12,26 @@
...
-            }
-        } else {
-            return "You failed to upload  because the file was empty.";
-        }
+			try {
+				file.transferTo(new File(servletContext.getRealPath("/images")
+						+ file.getOriginalFilename()));
+				return "images/" + file.getOriginalFilename();
+			} catch (Exception e) {
+				return "You failed to upload " + e.getMessage();
```
Comment (nantic) [inline HRTool/src/main/java/app/controllers/ImageUploadRestController.java]: Never return error message in this format.
The appropriate format would be: return new ResponseEntity<>("<ERROR_ENUM>", HttpStatus.BadRequest) or something close.

File: HRTool/src/main/webapp/app/partials/employeeDialog.html [hunk 10]
```diff
@@ -12,13 +12,10 @@
...
 			     <div flex offset = "15">
-					<div flow-init="{target: '/upload'  , testChunks:false, singleFile: 'true', uploadMethod: 'post', method: 'multipart', fileParameterName:'slika' }"
+					<div flow-init="{target: '/upload'  , testChunks:false, singleFile: 'true', uploadMethod: 'post', method: 'multipart', fileParameterName:'picture' }"
 					     flow-files-submitted="$flow.upload()"
 					     flow-file-success="imageUploaded($message, $file.size)"
 					     flow-file-added = "!!{png:1,gif:1,jpg:1,jpeg:1}[$file.getExtension()] && $file.size < 1024000">
-					    <div class="thumbnail" ng-hide="imageSwitch" >
-					      	<img src="images/Koala.jpg" height = "250" width = "250"/>
-					    </div>
 					    <div class="thumbnail" ng-show="imageSwitch">
```
Comment (nantic) [inline HRTool/src/main/webapp/app/partials/employeeDialog.html:19]: Use full variable names.
Ex. emp -> employee; currEmp -> currentEmployee

File: HRTool/src/main/java/app/controllers/ImageUploadRestController.java (hunk 5 above)
Comment (LjiljanaN) [inline HRTool/src/main/java/app/controllers/ImageUploadRestController.java]: I don't get this one. Where should I put those methods?

**Pull request closed without merging**